docs/ref/models/querysets.txt

```diff
@@ -3905,7 +3905,9 @@ by the aggregate.
 
     * Default alias: ``<field>__avg``
     * Return type: ``float`` if input is ``int``, otherwise same as input
-      field, or ``output_field`` if supplied
+      field, or ``output_field`` if supplied. If the queryset or grouping
+      is empty the value of ``default`` is returned, which is ``None`` by
+      default
```

Comment:
Hi, I think we disagree about this.

In the original case #34808 I wrote:

> And I'm also thinking that the "Return type" documentation for each individual function (Sum, Avg, etc) needs to mention the case where it may return None. While there is a note about empty querysets directly under the Aggregation functions header, I think the way documentation is used is that a lot of people scan the list of functions in the sidebar and click the one they're interested in. This scrolls them directly to the function documentation (e.g. for Sum) and they'll never see the empty queryset note.

I'm aware that including the return type for empty querysets feels repetitive and it feels more DRY to only mention it once up above. But if the only place we mention it is a place users don't see, I think that's a problem.

>I believe there is a responsibility on the reader's part to gather context from the surrounding text. 

Sure -- if the user knows to look. I think this case is more of an "unknown unknown". If you are implementing Sum for the first time, you're likely testing with sample querysets that are not empty. You won't realize you even need to worry about what an empty queryset returns until it happens to you -- which might not be for a quite a while. (And in my case, not until production.)

Reply:
Thank you @ebaumgartner for taking another look!

In my proposal, I balanced both things: your request to explicitly mention the return value when the queryset is empty, and being concise and clear.

> I'm aware that including the return type for empty querysets feels repetitive and it feels more DRY to only mention it once up above. But if the only place we mention it is a place users don't see, I think that's a problem.

The adding "If the queryset or grouping is empty, ``default`` is returned.", is quite explicit about what value (and its type) is returned in that case, which is *exactly* the value of `default`. The default for `default` is clearly shown in the same line of the signature of `Sum`, when it shows `class Sum(expression, output_field=None, distinct=False, filter=None, default=None, **extra)`.

So the reader should be able to immediately pick up that `None` is returned for empty querysets when they haven't passed a custom value for `default`.

> Sure -- if the user knows to look. I think this case is more of an "unknown unknown". If you are implementing Sum for the first time, you're likely testing with sample querysets that are not empty. You won't realize you even need to worry about what an empty queryset returns until it happens to you -- which might not be for a quite a while. (And in my case, not until production.)

The proposal only expects the reader to read the line describing the signature of the method they care about (and the "return type" sentence, of course).

Reply:
Thanks, I'd missed that default is now explicitly noted in the function definition.

Given that, I'm on board with your changes.